spring-kafka/src/test/java/org/springframework/kafka/listener/KafkaMessageListenerContainerTests.java

```diff
@@ -436,6 +443,61 @@ public void onMessage(ConsumerRecord<Integer, String> message) {
 		logger.info("Stop " + this.testName.getMethodName());
 	}
 
+	@Test
+	public void testRecordAck() throws Exception {
+		logger.info("Start record ack");
+		Map<String, Object> props = KafkaTestUtils.consumerProps("test6", "false", embeddedKafka);
+		DefaultKafkaConsumerFactory<Integer, String> cf = new DefaultKafkaConsumerFactory<>(props);
+		ContainerProperties containerProps = new ContainerProperties(topic6);
+		containerProps.setMessageListener((MessageListener<Integer, String>) message -> {
+			logger.info("record ack: " + message);
+		});
+		containerProps.setSyncCommits(true);
+		containerProps.setAckMode(AckMode.RECORD);
+		containerProps.setAckOnError(false);
+		KafkaMessageListenerContainer<Integer, String> container = new KafkaMessageListenerContainer<>(cf,
+				containerProps);
+		container.setBeanName("testAckOnError");
+		container.start();
+		ContainerTestUtils.waitForAssignment(container, embeddedKafka.getPartitionsPerTopic());
+		KafkaConsumer<?, ?> containerConsumer = spy(
```

Comment:
This `spy()` and subsequent `DFA` can be replaced by just one `spyOnConsumer()`.
And it really would be better to make any stubbing before `waitForAssignment()`.
Otherwise there is no guaranty that all the environment is proper for further verifications.

Will fix on merge.